crates/iceberg/src/spec/partition.rs

```diff
@@ -118,9 +118,63 @@ impl PartitionSpec {
     /// Turn this partition spec into an unbound partition spec.
     ///
     /// The `field_id` is retained as `partition_id` in the unbound partition spec.
-    pub fn to_unbound(self) -> UnboundPartitionSpec {
+    pub fn into_unbound(self) -> UnboundPartitionSpec {
         self.into()
     }
+
+    /// Check if this partition spec is compatible with another partition spec.
+    ///
+    /// Returns true if the partition spec is equal to the other spec with partition field ids ignored and
+    /// spec_id ignored. The following must be identical:
+    /// * The number of fields
+    /// * Field order
+    /// * Field names
+    /// * Source column ids
+    /// * Transforms
+    pub fn compatible_with(&self, other: &UnboundPartitionSpec) -> bool {
```

Comment:
NIT: How about renaming to `is_compatible_with`? This will align our naming style in existing APIs like `is_unpartitioned`.

Reply:
Sounds good - done in https://github.com/apache/iceberg-rust/pull/612/commits/64303ea21a26708b873a62f54a43b4d1e2265f94